### PR TITLE
Update flake8 and docs

### PR DESCRIPTION
**Pull request by nwu63**
## Purpose
- fixed flake8 issues
- updated docs layout due to #64
- added `__all__` to the module init files

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Documentation update
- [x] Maintenance update
- [ ] Other (please describe)

## Testing
N/A

## Checklist
_Put an `x` in the boxes that apply._

- [x] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

**Review by marcomangano — APPROVED** [review]
Looks good to me! The addition of `__all__` variable is needed for sphinx to access the classes in the subfolders, right?

**Comment by nwu63** [comment]
> Looks good to me! The addition of `__all__` variable is needed for sphinx to access the classes in the subfolders, right?

Exactly. They are also used if anyone were to use `from baseclasses.problems import *`, but that is heavily discouraged. They also prevent flake8 warnings of unused imports in the init file (even though we tend to blanket ignore those).

**Comment by marcomangano** [comment]
@sseraj or @eirikurj when you have time pls review and merge this

File: doc/BaseRegTest.rst [hunk 1]
```diff
@@ -1,12 +1,8 @@
-.. _BaseRegTest:
```
Comment (sseraj) [inline doc/BaseRegTest.rst:1]: Why remove the targets?

File: doc/index.rst [hunk 2]
```diff
@@ -57,3 +55,9 @@ Or from source, by cloning the repo and then running (from the repo root)::
 
    BaseRegTest
    regression_example
+
+.. toctree::
+   :maxdepth: 1
+   :caption: Utilities
+
+   utils
```
Comment (sseraj) [inline doc/index.rst:63]: Should there be an rst file for this?

File: doc/BaseRegTest.rst (hunk 1 above)
Comment (nwu63) [reply doc/BaseRegTest.rst:1]: I figured that people can just reference the class directly via `:class:BaseRegTest` or reference the page via the `:doc:` directive, so these targets are unnecessary.

File: doc/index.rst (hunk 2 above)
Comment (nwu63) [reply doc/index.rst:63]: Oh shoot I forgot to include the rst file in git, will push it up.

**Review by sseraj — APPROVED** [review]

**Pull request merged**